ci-operator/step-registry/baremetal/lab/pre/firewall/baremetal-lab-pre-firewall-commands.sh

```diff
@@ -13,6 +13,29 @@ SSHOPTS=(-o 'ConnectTimeout=5'
   -o LogLevel=ERROR
   -i "${CLUSTER_PROFILE_DIR}/ssh-key")
 
+setup_proxy() {
+  proxy="$(<"${CLUSTER_PROFILE_DIR}/proxy")"
+  cat <<EOF > "${SHARED_DIR}/proxy-conf.sh"
+  export HTTP_PROXY=${proxy}
+  export HTTPS_PROXY=${proxy}
+  export NO_PROXY="localhost,127.0.0.1"
+
+  export http_proxy=${proxy}
+  export https_proxy=${proxy}
+  export no_proxy="localhost,127.0.0.1"
+EOF
+}
+# shellcheck disable=SC2154
+if [ "${masters}" -eq 1 ]; then
```

Comment:
why when masters == 1?

Reply:
It seems you want to skip the LB when masters=1, but that's violating the architecture of the infra. We can bear having the additional LB even when SNO.

Reply:
This does not violate the architecture, it is how ABI SNO is designed to operate. An additional load balancer is not needed when installing an SNO cluster using ABI.

Reply:
Sure, none of the configurations strictly need that load balancer, but this is how the infra is expected to work: we rely on a single lb with a dedicated IP for each cluster. Only when the cluster is disconnected we use the (single instance) forward proxy (squid). 

It's not changing the ABI scenario, whereas using squid because the master node is not directly reachable does. This is hidden by the fact that we still need a proxy to reach via ipv6 on single stack... but for other reasons, not because it's an SNO.

Reply:
The LB work is to bridge the two networks: public and private, without needing a proxy. We only require the proxy because the prow clusters don't support ipv6 now.